### PR TITLE
preprocess: initial add

### PR DESCRIPTION
**Pull request by johannesring**

### CODE REVIEW

File: recipes/preprocess/meta.yaml [hunk 1]
```diff
@@ -0,0 +1,37 @@
+{% set version = "1.2.2" %}
+{% set git_rev = "d5ab9a0544cd2b15b0e8ae4f0b143b9047d1a54b" %}
+
+package:
+  name: preprocess
+  version: {{ version }}
+
+source:
+  git_url: https://github.com/hplgit/preprocess.git
+  git_rev: {{ git_rev }}
```
Comment (jakirkham) [inline recipes/preprocess/meta.yaml]: Can we please download a tarball instead? Maybe need to ask upstream to add some tags if we go this route.

File: recipes/preprocess/meta.yaml [hunk 2]
```diff
@@ -0,0 +1,37 @@
...
+
+build:
+  number: 0
+  script: python setup.py install --single-version-externally-managed --record record.txt
+
+requirements:
+  build:
+    - python
+    - setuptools
+    - future
```
Comment (jakirkham) [inline recipes/preprocess/meta.yaml]: Can we drop `future` from build? Guessing we only need this at run time.

**Comment by jakirkham** [comment]
Thoughts on this @johannesring ?

File: recipes/preprocess/meta.yaml (hunk 1 above)
Comment (johannesring) [inline recipes/preprocess/meta.yaml]: Sorry for not replying earlier. I did report an [issue](https://github.com/hplgit/preprocess/issues/8) for this upstream but never got a reply. I do not think it is likely that we will see any release tarballs or tags. Unfortunately, Hans Petter Langtangen, which maintained `preprocess`, passed away last week after long time illness.

**Comment by johannesring** [comment]
@jakirkham - please see comments above.

File: recipes/preprocess/meta.yaml (hunk 2 above)
Comment (jakirkham) [inline recipes/preprocess/meta.yaml]: Ok, that's unfortunate, but it makes sense. We can keep it then.

**Comment by johannesring** [comment]
@jakirkham - I decided to fork preprocess and maintain it myself for now. I have addressed the issues mentioned above.

**Comment by johannesring** [comment]
@jakirkham - I think the "Awaiting author contribution" label can be removed. Is it possible for me to remove it?

**Comment by jakirkham** [comment]
What do we think @conda-forge/core ? Normally would be against packaging a fork. However, there are extenuating circumstances here.

**Comment by scopatz** [comment]
Is it possible to apply the diff between the fork and the original as a patch here? Or is that too much? Is your fork the only maintained version now?

**Comment by jakirkham** [comment]
Based on the discussion in issue ( https://github.com/hplgit/preprocess/issues/8 ). It sounds like forking was the only reasonable option as the main version is unmaintained.

**Comment by scopatz** [comment]
I think it is fine then.

**Comment by johannesring** [comment]
@jakirkham, @scopatz - "We" aka @KGHustad decided to move the `preprocess` fork to the new [doconce](https://github.com/doconce/) organization. I have updated this PR for this move now.

File: recipes/preprocess/meta.yaml [hunk 3]
```diff
@@ -0,0 +1,46 @@
...
+  run:
+    - python
+    - future
+
+test:
+  imports:
+    - preprocess
+
+about:
+  home: https://github.com/johannesring/preprocess
```
Comment (scopatz) [inline recipes/preprocess/meta.yaml]: Should this URL change?

Comment (johannesring) [inline recipes/preprocess/meta.yaml]: Yes, thanks - I forgot about that. Will fix it tomorrow.

Comment (johannesring) [inline recipes/preprocess/meta.yaml]: Fixed now.

**Comment by johannesring** [comment]
Thanks @scopatz!

**Pull request merged**